spaceship/lib/spaceship/client.rb

```diff
@@ -202,15 +202,18 @@ def team_id=(team_id)
       #     invalid content provider id
       #
       available_teams = teams.collect do |team|
-        (team["contentProvider"] || {})["contentProviderId"]
+        {
+          team_id: (team["contentProvider"] || {})["contentProviderId"],
+          team_name: (team["contentProvider"] || {})["name"]
+        }
       end
 
       result = available_teams.find do |available_team_id|
```

Comment:
Now that `available_teams` is an array of hashes is this going to work appropriately? Is there any chance you could add a test for this?

Reply:
Yeah I don't see this variable being used anywhere else. The scope of the variable is local so it shouldn't affect any code outside the function.

Reply:
Line 212 here is going to fail because it is trying to compare the `team_id` to the `available_team_id` but with this change, `available_team_id` is a hash right?

Reply:
Good catch. I fixed it.

Reply:
@ohayon do you know how can I get rid of all those commits in my branch?

Reply:
You should be able to do a `git rebase -i HEAD~28` and then select fixup for all but the important commit. Also, maybe an easier way would be to just pull master, checkout a new branch and remake your changes. I think that will be the faster option.

Reply:
@ohayon Thanks, I will do that for the next pull request.